Tabloid/Controllers/UserProfileController.cs

```diff
@@ -60,5 +65,45 @@ public IActionResult Post(UserProfile userProfile)
                 new { firebaseUserId = userProfile.FirebaseUserId },
                 userProfile);
         }
+
+        [HttpPut("deactivate/{id}")]
+        public IActionResult Deactivate(int id, UserProfile profile)
+        {
+            try
+            {
+                if(id != profile.Id)
+                {
+                    return BadRequest();
+                }
+                var currentUser = GetCurrentUserProfile();
+                if (currentUser.UserTypeId == 1)
+                {
+                    _userProfileRepository.Deactivate(id);
+                    return NoContent();
+                }
+                else
+                {
+                    return Unauthorized();
+                }
+                
+            }
+            catch (Exception ex)
```

Comment:
Just out of curiosity, what cases do you imagine this catching?

Reply:
I can't really remember what my thought process was on that try catch block, but I think I added it before I added the if (id != profile.id) condition. One possible case it could catch is where the id does equal to profile.Id but someone else deactivated that user in between the confirmation page and clicking confirm, so that the user is actually not in the database? Though I'm not sure what happens when the where clause in the sql query matches nothing.

Reply:
Ok, it's not hurting anything. resolving.